Anna Nenarokova
============

<a.nenarokova@gmail.com>  

[Github profile](https://github.com/AnnaNenarokova)
[Google Scholar profile](https://scholar.google.com/citations?user=IpR3R6IAAAAJ&hl=en)

## Research interests

* Genomics and metagenomics of microorganisms
* Genome evolution
* Diversity of microorganisms
* Origin and evolution of eukaryotes
* Early evolution of life
* Origin and evolution of the genetic code

Skills
-----

* Proficient in operating Unix-based workstations and HPC clusters
* Programming: scripting (Bash, Python), creating and managing pipelines; code review and modification (R, C, C++, JavaScript etc.)
* Preparing of raw short and long NGS reads: debarcoding, trimming, merging, filtering etc.
* Genome (metagenome) and transcriptome (metatranscriptome) assembly, filtering, decontamination
* Read mapping and polymorphism analysis
* Structural and functional annotation of prokaryotic and eukaryotic genomes
* Genome alignment and comparative genomics methods
* Phylogenetics and phylogenomics analyses
* Planning and implementing bioinformatics research
* Academic writing and preparing data for publications
* Teaching and mentoring in life sciences and bioinformatics
* Presenting to an audience with a varied background

Academic career
---------

2021/9-present
:        **Postdoctoral researcher**; University of Bristol, United Kingdom
    
    * Involved in the Moore–Simons Project on the Origin of the Eukaryotic Cell "Resolving the prokaryote-to-eukaryote transition using new phylogenomic models of endosymbiosis".
    Advisors - Dr. Tom Williams <tom.a.williams@bristol.ac.uk>, Prof. Philip Donoghue <Phil.Donoghue@bristol.ac.uk>

2020/3 - 2021/9
:       **Postdoctoral researcher**; University of South Bohemia, České Budějovice, Czech Republic

2015-2020
:       **Ph.D. in Genetics and Molecular Biology**; University of South Bohemia, Czech Republic
    * Ph.D. thesis: “Genomics of *Blastocrithidia*, the trypanosomatid with all three stop codons reassigned”

2009-2015
:       **Bachelor and Master of Science in Genetics**; Lomonosov Moscow State University, Russia

    * Graduated with honors (cum laude) 
    * Master’s thesis: “Polymorphism analysis of NBS LRR class resistance genes in wheat varieties using high-throughput sequencing”  
    * Bachelor’s thesis: “Adaptation process in the CRISPR immunity system”
 

Professional Experience
----------

2021/9-present
:       **Postdoctoral researcher**; University of Bristol, United Kingdom
        **Advisors: Drs. Tom Williams, Davide Pisani and Philip Donoghue**
        * Working on identifying the prokaryotic origins of eukaryotic genes
        * Working on the molecular clock analyses for the tree of life
        * Inferring and analysing phylogenetic trees of eukaryotes and prokaryotes
        * Developing scripts and pipelines for phylogenomic analyses
        * Co-supervising and mentoring students in the group
        * Assembled a representative dataset of eukaryotic genomes and marker genes for phylogenetic analysis

2021/1 - 2021/8
:       **Postdoctoral researcher**; University of South Bohemia, České Budějovice, Czech Republic
        **Advisors: Drs. Václav Hypša and Eva Nováková**
        * Provided support and supervision in bioinformatics for the group
        * Designed scripts and pipelines for metagenomic analyses for the group
        * Participated in the assembly, filtering, and analysis of the genomes of parasitic insects and their bacterial endosymbionts

2020/3 - 2021/1
:       **Postdoctoral researcher**; Biology Center CAS, České Budějovice, Czech Republic
        **Advisor: Dr. Astrid Holzer**
        * Provided support and supervision in bioinformatics for the group
        * Participated in the assembly, filtering and analysis of the genome and the transcriptome of the myxozoan *Sphaerospora molnari*, analysed horizontal gene transfers in its genome

2019/1 - 2019/4
:       **Visiting Postgraduate Researcher**; Bristol University, Bristol, UK
        **Advisor: Dr. Tom Williams**
        * [Analysed](https://link.springer.com/article/10.1186/s12915-023-01563-9) horizontal gene transfers in *Diplonema papillatum* genome

2015-2020
:       **Ph.D. Researcher**; Institute of Parasitology, České Budějovice, Czech Republic
        **Advisor: Dr. Julius Lukeš**
        * [Discovered](https://www.nature.com/articles/s41586-022-05584-2) the effect of Trp-tRNA mutations that was experimentally proven in trypanosomatids and yeast
        * [Assembled and analyzed](https://static-content.springer.com/esm/art%3A10.1038%2Fs41586-022-05584-2/MediaObjects/41586_2022_5584_MOESM1_ESM.pdf) the genome of *Blastocrithidia nonstop*, [proposed](https://static-content.springer.com/esm/art%3A10.1038%2Fs41586-022-05584-2/MediaObjects/41586_2022_5584_MOESM1_ESM.pdf) an evolutionary scenario explaining its unique features
        * Designed [an algorithm for structural genome annotation with ambiguous stop codons](https://zenodo.org/record/7116082) and supervised the software developement
        * Coordinated the work of experimental biologists and bioinformaticians
        * [Discovered](https://mbio.asm.org/content/10/4/e01541-19.abstract) the correlation between the presence of the C-NHEJ DNA repair pathway in different groups of eukaryotes and their genome features
        * Participated in [the analysis](https://journals.plos.org/plospathogens/article?id=10.1371/journal.ppat.1006679) of the mitochondrial proteome of *T. brucei*
        * Assembled and annotated [the genome](https://journals.asm.org/doi/full/10.1128/mbio.01606-21) of trypanosomatid *Novymonas esmeraldas* and its [bacterial endosymbiont](https://www.frontiersin.org/articles/10.3389/fmicb.2017.01940/full) *Ca.* Pandoraea novymonadis
        * Participated in *Euglena gracilis* [genome](https://bmcbiol.biomedcentral.com/articles/10.1186/s12915-019-0626-8) and [organellar proteome](https://academic.oup.com/mbe/advance-article/doi/10.1093/molbev/msaa061/5803078) projects: annotated and analyzed mitochondrial proteome - **collaboration with Dr. Mark Field's group, University of Dundee, UK**

2014-2015
:       **Master's Researcher**; Vavilov Institute of General Genetics, Moscow, Russia
        **Advisors:  Drs. Artem Kasianov and Alexey Penin**
        * Used NGS data to study disease resistance genes of wheat
        * Examined polymorphism in disease resistance genes of wheat varieties
        * Developed a pipeline in Python and Bash to handle a large amount of low quality NGS data on an HPC cluster

2013-2014
:       **Undergraduate Researcher**; Institute of Gene Biology, Moscow, Russia 
        **Advisor: Dr. Konstantin Severinov**
        * Examined genomes of microcine resistant *E. coli* strains to reveal causes for resistance
        * Used NGS to study bacterial CRISPR immunity
        * Developed utilities in R and Python to automate the analysis of novel CRISPR spacers in *E. coli*
        * Constructed plasmids and conducted experiments to access CRISPR adaptation in *E. coli*


Publications 
------------

**H-index 9**

* Donoghue P.C.J., Kay C., Spang A., Szöllősi G., **Nenarokova A.**, Moody E.R.R., Pisani D., Williams T.A. *Curr Biol.* 2023 Sep 11;33(17):R919-R929. Defining eukaryotes to dissect eukaryogenesis. [doi.org:10.1016/j.cub.2023.07.048](https://doi.org/10.1016/j.cub.2023.07.048)
* Valach M., Moreira S., Petitjean C., Benz A., Butenko A., Flegontova O., **Nenarokova A.**, Prokopchuk G., Batstone T., Lapébie P., Lemogo L., Sarrasin M., Stretenowich P., Tripathi P., Yazaki E., Nara T., Henrissat B.,Lang B.F., Gray M.W., Williams T.A., Lukeš J., Burger G. Recent expansion of metabolic versatility in *Diplonema papillatum*, the model species of a highly speciose group of marine eukaryotes. *BMC Biol.* 2023 May 4;21(1):99. [doi.org:10.1186/s12915-023-01563-9](https://doi.org/10.1186/s12915-023-01563-9)
* Kachale A., Pavlíková Z., **Nenarokova A.** (**shared first co-authorship**), Roithová A., Durante I.M., Miletínová P., Záhonová K., Nenarokov S., Votýpka J., Horáková E., Ross R.L., Yurchenko V., Beznosková P., Paris Z., Valášek L.S., Lukeš J. Short tRNA anticodon stem and mutant eRF1 allow stop codon reassignment. *Nature* 2023 Jan;613(7945):751-758. [doi.org:10.1038/s41586-022-05584-2](https://doi.org/10.1038/s41586-022-05584-2) **Journal cover**
* Paris Z., Svobodová M., Kachale A., Horáková E., **Nenarokova A.**, Lukeš J., 2021. A mitochondrial cytidine deaminase is responsible for C to U editing of tRNAᵀʳᵖ to decode the UGA codon in *Trypanosoma brucei*. *RNA Biol.* 2021 Oct 15;18(sup1):278-286. [doi.org:10.1080/15476286.2021.1940445](https://doi.org/10.1080/15476286.2021.1940445).
* Zakharova A., Saura A., Butenko A., Podešvová L., Warmusová S., Kostygov A.Y., **Nenarokova A.**, Lukeš J., Opperdoes F.R., Yurchenko V., 2021. A New Model Trypanosomatid, *Novymonas esmeraldas*: Genomic Perception of Its “*Candidatus* Pandoraea novymonadis” Endosymbiont. *mBio* 2021 Aug 31;12(4):e0160621 [doi.org:10.1128/mbio.01606-21](https://journals.asm.org/doi/full/10.1128/mbio.01606-21)
* Hammond M.J., **Nenarokova A.**, Butenko A., Zoltner M., Lacová Dobáková E., Field M.C., Lukeš J., 2020. A uniquely complex mitochondrial proteome from *Euglena gracilis*. *Mol Biol Evol.* 2020 Aug 1;37(8):2173-2191.[doi:10.1093/molbev/msaa061](https://academic.oup.com/mbe/advance-article/doi/10.1093/molbev/msaa061/5803078).
* Kosakyan A., Alama-Bermejo G., Bartošová-Sojková P., Born-Torrijos A., Šíma R., **Nenarokova A.**, Eszterbauer E., Bartholomew J., Holzer A.S., 2019. Selection of suitable reference genes for gene expression studies in myxosporean (Myxozoa, Cnidaria) parasites. *Sci Rep.* 2019 Oct 21;9(1):15073. [doi:10.1038/s41598-019-51479-0](https://www.nature.com/articles/s41598-019-51479-0).
* **Nenarokova A.**, Záhonová K., Krasilnikova M., Gahura O., McCulloch R., Zíková A., Yurchenko V., Lukeš J., 2019. Causes and effects of loss of classical nonhomologous end joining pathway in parasitic eukaryotes. *mBio* 2019 Jul 16;10(4):e01541-19. [doi:10.1128/mBio.01541-19](https://mbio.asm.org/content/10/4/e01541-19.abstract). **Editor's pick** 
* Ebenezer T., Zoltner M., Burrel A., **Nenarokova A.**, Novák Vanclová A.M.G., Prasad B., Soukal P., Santana-Molina C., O’Neill E., Nankissoor N.N., Vadakedath N., Daiker V., Obado S., Jackson A.P., Devos D., Lukeš J., Lebert M., Vaughan S., Hampl V., Carrington M., Ginger M.L., Dacks J.B., Kelly S., Field M.C., 2018. Transcriptome, proteome and draft genome of *Euglena gracilis*. *BMC Biol.* 2019 Feb 7;17(1):11. [doi:10.1186/s12915-019-0626-8](https://bmcbiol.biomedcentral.com/articles/10.1186/s12915-019-0626-8).
* Harmer J., Yurchenko V., **Nenarokova A.**, Lukeš J., Ginger M.L., 2018. Farming, slaving and enslavement: histories of endosymbioses during kinetoplastid evolution. *Parasitology* 2018 Sep;145(10):1311-1323. [doi:10.1017/S0031182018000781](https://www.ncbi.nlm.nih.gov/pubmed/29895336). **Journal cover**
* Zíková A., Verner Z., **Nenarokova A.**, Michels P.A.M., Lukeš J., 2017. A paradigm shift: The mitoproteomes of procyclic and bloodstream *Trypanosoma brucei* are comparably complex. *PLoS Pathog.* 2017 Dec 21;13(12):e1006679. [doi:10.1371/journal.ppat.1006679](https://journals.plos.org/plospathogens/article?id=10.1371/journal.ppat.1006679).
* Schwelm A., Badstöber J., Bulman S., Desoignies N., Etemadi M., Falloon R. E., Gachon C. M. M., Legreve A., Lukeš J., Merz U., **Nenarokova A.**, Strittmatter M., Sullivan B. K. and Neuhauser S., 2018. Not in your usual Top 10: protists that infect plants and algae. *Mol Plant Pathol.* 2018 Apr;19(4):1029-1044.  [doi:10.1111/mpp.12580](https://bsppjournals.onlinelibrary.wiley.com/doi/full/10.1111/mpp.12580).
**Journal cover**
* Kostygov A.Y., Butenko A., **Nenarokova A.**, Tashyreva D., Flegontov P., Lukeš J. and Yurchenko V., 2017. Genome of *Ca.* Pandoraea novymonadis, an endosymbiotic bacterium of the trypanosomatid *Novymonas esmeraldas*. *Front Microbiol.* 2017 Oct 4;8:1940. [doi:10.3389/fmicb.2017.01940](https://www.frontiersin.org/articles/10.3389/fmicb.2017.01940/full).

Awards and Scholarships 
------

* University of South Bohemia travel grant for scientific stay, 2019
* Holz-Conner award of International Society of Protistologists, 2018
* University of South Bohemia Academic Excellence Scholarship, 2016-2020
* Moscow State University Academic Excellence Scholarship, 2009-2015
* Moscow Government’s Award for Talented Students, 2008

Grants
------

Participant:

* Russian Science Foundation grant “Amoebozoa of the World Ocean: phylogeny, community composition and global distribution patterns”. Principal investigator: A.A. Kudryavtsev, **2020-2022**.
* Czech Grant Agency, 18-15962S "Trypanosomatids with in-frame sense stop codons: dissecting molecular mechanisms behind the unique ambiguity of the genetic code". Principal investigators: V. Yurchenko, J. Lukeš, **2018-2020**.
* European Commission Operational Programme Research, Development and Education (Call: Excellent Research), "Centre for research of pathogenicity and virulence of parasites". Principal investigators: J. Tachezy, P. Doležal, R. Šuťák, V. Hampl, P. Volf, P. Horák, T. Drmota, J. Lukeš, P. Kopáček, M. Oborník, A. Zíková, Z. Paris, M. Kolisko, M. Eliáš, V. Yurchenko, **2018-2022**.
* ERC CZ grant LL1601 “Comprehensive study of diplonemids: emerging key players in the oceans”. Principal investigator: J. Lukeš, **2017-2021**.
* Russian Science Foundation grant 14-14-00988 “The function and evolution of CRISPR/Cas system”. Principal investigator: K. Severinov, **2014-2017**.

Conferences
------
* **Nenarokova A.**, Dombrowski N., Morel B., Mahendrarajah T., Schrempf D., Szánthó L., Donoghue P.C.J., Pisani D., Szöllősi G.J., Spang A., Williams T.A. A roadmap to resolve the prokaryote-to-eukaryote transition. // Microbial Evolution Network Ireland Meeting, Cork, Ireland, 2023. Oral presentation
* **Nenarokova A.**, Dombrowski N., Morel B., Mahendrarajah T., Schrempf D., Szánthó L., Donoghue P.C.J., Pisani D., Szöllősi G.J., Spang A., Williams T.A. Resolving the prokaryote-to-eukaryote transition using new phylogenomic models of endosymbiosis. // Microbiology Society Annual Conference, Birmingham, United Kingdom, 2023. Poster
* **Nenarokova A.**, Dombrowski N., Morel B., Szánthó L., Donoghue P.C.J., Pisani D., Szöllősi G.J., Spang A., Williams T.A. A roadmap to resolve the prokaryote-to-eukaryote transition. // Progress and Development in Phylogenetic Methods, Oslo, Norway, 2023. Oral presentation
* **Nenarokova A.**, Dombrowski N., Morel B., Szánthó L., Donoghue P.C.J., Pisani D., Szöllősi G.J., Spang A., Williams T.A. Resolving eukaryogenesis using phylogenetics. // UK Archaea Workshop, London, United Kingdom, 2023. Oral presentation
* **Nenarokova A.**, Dombrowski N., Szánthó L., Donoghue P.C.J., Pisani D., Szöllősi G.J., Spang A., Williams T.A. A roadmap to resolve the prokaryote-to-eukaryote transition. // Moore-Simons Project on the Origin of the Eukaryotic Cell Annual Meeting, 2022. Online presentation
* **Nenarokova A.**, Záhonová K., Nenarokov S., Paris Z., Yurchenko V., Lukeš J. *Blastocrithidia*, the trypanosomatid with all three stop codons reassigned. // Joint meeting of the Phycological Society of America and the International Society of Protistologists, Vancouver, Canada, 2018. Oral presentation
* **Nenarokova A.**, Záhonová K., Nenarokov S., Yurchenko V., Lukeš J. Genomics of *Blastocrithidia*, the trypanosomatid with all three stop codons reassigned. // XXII Meeting of the International Society for Evolutionary Protistology, Droushia, Cyprus, 2018. Oral presentation
* **Nenarokova A.**, Záhonová K., Nenarokov S., Paris Z., Boudová M.,  Skalický T., Yurchenko V., Lukeš J. *Blastocrithidia*, a trypanosomatid with all three stop codons reassigned. // EMBO meeting, Sant Feliu de Guixols, Spain, 2017. Poster. **Best Poster Award**
* **Nenarokova A.**, Záhonová K., Nenarokov S., Paris Z., Boudová M.,  Skalický T., Yurchenko V., Lukeš J. *Blastocrithidia*, a trypanosomatid with all three stop codons reassigned. // International Congress of Protistologists, Prague, Czech Republic, 2017. Oral presentation
* **Nenarokova A.**, Dobáková E., Flegontov P., Ebenezer T., Zolner M., Field M., Lukeš J. Functional annotation of *Euglena gracilis* mitochondrial proteome. // British Society for Parasitology Trypanosomiasis & Leishmaniasis Seminar, České Budějovice, Czech Republic, 2016. Poster
* **Nenarokova A.**, Flegontov P., Field  M.C., Lukeš J. Functional annotation of *Euglena gracilis* mitochondrial proteome. // TryTax2 Meeting, Ostrava, Czech Republic, 2016. Poster
* **Nenarokova A.**, Flegontov P., Field  M.C., Lukeš J. Functional annotation of *Euglena gracilis* mitochondrial proteome. //  Protodays, Bítov, Czech Republic, 2016. Poster. **Best Poster Award**
* **Nenarokova A.A.** Polymorphism analysis of NBS LRR class resistance genes in wheat varieties using high-throughput sequencing. // Bioinformatics Summer School, St. Petersburg, Russia, 2015. Oral presentation
* **Nenarokova A.A.**, Strotskaya A.V. Adaptation process in the CRISPR immunity system. // The 18th Pushchino School-Conference of Young Scientists “Biology – The Science of The XXI Century”, Pushchino, Russia, 2014. Poster. **Best Poster Award**
* **Nenarokova A.A.** Experimental system for studying the Effects of CRISPR/Cas genetic immunity system of *Escherichia coli* against lytic bacteriophages. //XXI Undergraduate and Postgraduate Student International Conference "Lomonosov",  Moscow, Russia, 2014. Poster

Peer Review
------

* Ad hoc reviewer for *Scientific Reports* and MDPI journals

Attended Workshops and Courses
-----

* [Success beyond the lab](https://www.successbeyondthelab.com/course-boost-your-career) - online career course focusing on soft skills for scientists, 2020 
* Workshop on phylogenomics [evomics.org](http://www.evomics.org), Český Krumlov, Czech Republic, 2017
* Elixir Excelerate workshop, Prague, 2016
* III Bioinformatics Summer School, St. Petersburg, Russia, 2015. Project: “Analysis of *Homo sapiens* Reference Genome”
* Future Biotech Winter School, Zvenigorod, Russia, 2015
* Future Biotech Autumn Intensity, Moscow, Russia, 2014
* II Bioinformatics Summer School, St. Petersburg, Russia, 2014. Project: “Genomic Analysis of Pathogenic Regions in *E. coli* O104:H4”

Teaching
-----

* [Volunteering as an online tutor for Ukrainian high school students](https://tutoringwithoutborders.org/) in biology and English language; 2022-present
* Organizer and lecturer of the “Introduction to programming for biologists” seminar at the Institute of Parasitology, Biology Centre CAS; 2020-2021
* Teaching assistant to the “Introduction to Bioinformatics” course at the University of South Bohemia; 2018.
* Bioinformatics mentoring for biologists; 2017-present
* Biology tutoring for high school students; 2010-2012

Outreach
-----

* As a volunteer media manager, I assisted in organizing the [SciFilm Hackathon at Bristol University](https://scifilmit.com/bristol-2023/), which aimed to produce science-based films for the general public. These films are distributed via [YouTube](https://www.youtube.com/watch?v=WTqmqZ5ZeB4&list=PLwS0cZFRgFCOLehz4_mnVlIFfh80u9QDS).
* Delivered several lectures on life sciences for a general audience, such as "Evolution of Complexity: Why fitness is not enough?" 20.06.2019, Prague, Czech Republic. The recording of the lecture is available [here](https://drive.google.com/file/d/1N6JVKUM1VbwXJA2UHTfuO4CQuc8jFEV2/view?usp=sharing). 
* I maintain active personal social media profiles with a focus on life sciences and STEM. You can follow me on [Facebook](https://www.facebook.com/anna.nenarokova/) and [X/Twitter](https://twitter.com/ANenarokova)

Languages
-----

* **Russian** - native
* **English** - full proficiency, C1 IELTS certification
* **Czech** - conversational, reading professional literature (A2)
* **French** - basic
* **German** - basic

References
-----

1. [Dr. Tom Williams](https://scholar.google.com/citations?user=HP3j4g0AAAAJ&hl=en)

* Position: [Associate Professor in Molecular Evolution](https://research-information.bris.ac.uk/en/persons/tom-williams)
* Institution: School of Biological Sciences, University of Bristol
* Email: <tom.a.williams@bristol.ac.uk>
* Relationship: Current supervisor

2. [Prof. Philip Donoghue](https://scholar.google.com/citations?user=VdQriAIAAAAJ&hl=en)

Position: [Professor of Palaeobiology](https://www.bristol.ac.uk/people/person/Philip-Donoghue-4fd2d6ef-3986-4566-97e4-7ae7db296525/)
Institution: School of Earth Sciences, University of Bristol
Email: <Phil.Donoghue@bristol.ac.uk>
Relationship: Current supervisor