Anna Nenarokova
============

> Institute of Parasitology, Biology Centre CAS  
> Branišovská 31, 370 05, České Budějovice, Czech Republic  
> (+420) 702 675 633  
> <a.nenarokova@gmail.com>  
> [Anna's Github](https://github.com/AnnaNenarokova)  

---

Current Position
---------

2020/3-present
:   **Postdoctoral researcher**; Institute of Parasitology,  Biology Center CAS and University of South Bohemia, České Budějovice, Czech Republic 

Education
---------

2015-2020
:   **Ph.D. in Genetics and Molecular Biology**; University of South Bohemia, Czech Republic

    * Ph.D. thesis: “Genomics of *Blastocrithidia*, the trypanosomatid with all three stop codons reassigned”

2009-2015
:   **Bachelor and Master of Science in Genetics**; Lomonosov Moscow State University, Russia

    * Graduated with honors (cum laude)
    
    * Master’s thesis: “Polymorphism analysis of NBS LRR class resistance genes in wheat varieties using high-throughput sequencing”
    
    * Bachelor’s thesis: “Adaptation process in the CRISPR immunity system”
 

Professional Experience
----------

2021/1 - present
:   **Postdoctoral researcher**; University of South Bohemia, České Budějovice, Czech Republic

    **Advisors: Drs. Václav Hypša <vacatko@prf.jcu.cz> and Eva Nováková <novaeva@paru.cas.cz> **

    * I provide support and supervision in bioinformatics for the group
    * I participate in the assembly and analysis of the genomes of blood-sucking insects and their bacterial endosymbionts

2020/3 - present
:   **Postdoctoral researcher**; Biology Center CAS, České Budějovice, Czech Republic

    **Advisor: Dr. Zdeněk Paris** <parda@paru.cas.cz>

    * I am analyzing the patterns of the distribution of queuosine codons in *Trypanosoma brucei* genes

    **Advisor: Dr. Astrid Holzer** <astrid.holzer@paru.cas.cz>

    * I provide support and supervision in bioinformatics for the group
    * I participate in the assembly and analysis of the genome of the myxozoan *Sphaerospora molnari*


2015-2020
:   **Ph.D. Researcher**; Institute of Parasitology, České Budějovice, Czech Republic

    **Advisor: Dr. Julius Lukeš** <jula@paru.cas.cz>

    * Assembled and analyzed the genome of *Blastocrithidia*
    * Designed an algorithm for structural genome annotation with ambiguous stop codons 
    * Coordinated the work of experimental biologists and bioinformaticians 
    * Predicted the effect of Trp-tRNA mutations that was experimentally proven in trypanosomatids and yeast
    * [Discovered](https://mbio.asm.org/content/10/4/e01541-19.abstract) the correlation between the presence of the C-NHEJ DNA repair pathway in different groups of eukaryotes and their genome features
    * Participated in [the analysis](https://journals.plos.org/plospathogens/article?id=10.1371/journal.ppat.1006679) of the mitochondrial proteome of *T. brucei*
    * Assembled and annotated [the genome](https://www.frontiersin.org/articles/10.3389/fmicb.2017.01940/full) of trypanosomatid *Novymonas esmeraldas* and its endosymbiont *Ca.* Pandoraea novymonadis
    * Participated in *Euglena gracilis* [genome sequencing](https://bmcbiol.biomedcentral.com/articles/10.1186/s12915-019-0626-8) and [organellar proteome](https://academic.oup.com/mbe/advance-article/doi/10.1093/molbev/msaa061/5803078) projects: annotated and analyzed mitochondrial proteome - **collaboration with Dr. Mark Field's group, University of Dundee, UK**

2014-2015
:   **Master's Researcher**; Vavilov Institute of General Genetics, Moscow, Russia

    **Advisors:  Dr. Artem Kasianov, Dr. Alexey Penin**

    * Used NGS data to study disease resistance genes of wheat
    * Examined polymorphism in disease resistance genes of wheat varieties
    * Developed a pipeline in Python and Bash to handle a large amount of low quality NGS data on a computer cluster

2013-2014
:   **Undergraduate Researcher**; Institute of Gene Biology, Moscow, Russia 

    **Advisor: Dr. Konstantin Severinov**

    * Examined genomes of microcine resistant *E. coli* strains to reveal causes for resistance
    * Used NGS to study bacterial CRISPR immunity
    * Developed utilities in R and Python to automate the analysis
    * Constructed plasmids and conducted experiments to access CRISPR adaptation in *E. coli*


Internships
--------------------

January-April 2019

:   **Visiting Postgraduate Researcher**; Bristol University, Bristol, UK

    **Advisor: Dr. Tom Williams**

    * Analysed horizontal gene transfers in *Diplonema papillatum* genome

Publications 
------------

**H-index 5**

* Hammond M.J., **Nenarokova A.**, Butenko A., Zoltner M., Lacová Dobáková E., Field M.C., Lukeš J., 2020. A uniquely complex mitochondrial proteome from *Euglena gracilis*. *Mol Biol Evol.*, Mar 11; msaa061; [doi:10.1093/molbev/msaa061](https://academic.oup.com/mbe/advance-article/doi/10.1093/molbev/msaa061/5803078).
* Kosakyan A., Alama-Bermejo G., Bartošová-Sojková P., Born-Torrijos A., Šíma R., **Nenarokova A.**, Eszterbauer E., Bartholomew J., Holzer A.S., 2019. Selection of suitable reference genes for gene expression studies in myxosporean (Myxozoa, Cnidaria) parasites, 2019. *Scientific Reports* 9(1); [doi:10.1038/s41598-019-51479-0](https://www.nature.com/articles/s41598-019-51479-0).
* **Nenarokova A.**, Záhonová K., Krasilnikova M., Gahura O., McCulloch R., Zíková A., Yurchenko V., Lukeš J., 2019. Causes and effects of loss of classical nonhomologous end joining pathway in parasitic eukaryotes. *mBio* 10(4), e01541-19; [doi:10.1128/mBio.01541-19](https://mbio.asm.org/content/10/4/e01541-19.abstract). **Editor's pick** 
* Ebenezer T., Zoltner M., Burrel A., **Nenarokova A.**, Novák Vanclová A.M.G., Prasad B., Soukal P., Santana-Molina C., O’Neill E., Nankissoor N.N., Vadakedath N., Daiker V., Obado S., Jackson A.P., Devos D., Lukeš J., Lebert M., Vaughan S., Hampl V., Carrington M., Ginger M.L., Dacks J.B., Kelly S., Field M.C., 2018. Transcriptome, proteome and draft genome of *Euglena gracilis*. *BMC Biology* 17(1), 1; [doi:10.1186/s12915-019-0626-8](https://bmcbiol.biomedcentral.com/articles/10.1186/s12915-019-0626-8).
* Harmer J., Yurchenko V., **Nenarokova A.**, Lukeš J., Ginger M.L., 2018. Farming, slaving and enslavement: histories of endosymbioses during kinetoplastid evolution. *Parasitology* 145, 1311–1323. [doi:10.1017/S0031182018000781](https://www.ncbi.nlm.nih.gov/pubmed/29895336). **Journal cover**
* Zíková A., Verner Z., **Nenarokova A.**, Michels P.A.M., Lukeš J., 2017. A paradigm shift: The mitoproteomes of procyclic and bloodstream *Trypanosoma brucei* are comparably complex. *PLoS Pathogens* 13(12), e1006679; [doi:10.1371/journal.ppat.1006679](https://journals.plos.org/plospathogens/article?id=10.1371/journal.ppat.1006679).
* Schwelm A., Badstöber J., Bulman S., Desoignies N., Etemadi M., Falloon R. E., Gachon C. M. M., Legreve A., Lukeš J., Merz U., **Nenarokova A.**, Strittmatter M., Sullivan B. K. and Neuhauser S., 2018. Not in your usual Top 10: protists that infect plants and algae. *Molecular Plant Pathology* 19(4); [doi:10.1111/mpp.12580](https://bsppjournals.onlinelibrary.wiley.com/doi/full/10.1111/mpp.12580).
**Journal cover**
* Kostygov A.Y., Butenko A., **Nenarokova A.**, Tashyreva D., Flegontov P., Lukeš J. and Yurchenko V., 2017. Genome of *Ca.* Pandoraea novymonadis, an endosymbiotic bacterium of the trypanosomatid *Novymonas esmeraldas*. *Frontiers Microbiology* 8:1940;  
[doi:10.3389/fmicb.2017.01940](https://www.frontiersin.org/articles/10.3389/fmicb.2017.01940/full).

Peer Review
------

* Ad hoc reviewer for *Scientific Reports*

Awards and Scholarships 
------

* University of South Bohemia travel grant for scientific stay, 2019
* Holz-Conner award, 2018
* University of South Bohemia Academic Excellence Scholarship, 2016-present
* Moscow State University Academic Excellence Scholarship, 2009-2015
* Moscow Government’s Award for Talented Students, 2008

Grants
------

Participant:

* Russian Science Foundation grant “Amoebozoa of the World Ocean: phylogeny, community composition and global distribution patterns”. Principal investigator: A.A. Kudryavtsev, **2020-2022**.
* Czech Grant Agency, 18-15962S "Trypanosomatids with in-frame sense stop codons: dissecting molecular mechanisms behind the unique ambiguity of the genetic code". Principal investigators: V. Yurchenko, J. Lukeš, **2018-2020**.
* European Commission Operational Programme Research, Development and Education (Call: Excellent Research), "Centre for research of pathogenicity and virulence of parasites". Principal investigators: J. Tachezy, P. Doležal, R. Šuťák, V. Hampl, P. Volf, P. Horák, T. Drmota, J. Lukeš, P. Kopáček, M. Oborník, A. Zíková, Z. Paris, M. Kolisko, M. Eliáš, V. Yurchenko, **2018-2022**.
* ERC CZ grant LL1601 “Comprehensive study of diplonemids: emerging key players in the oceans”. Principal investigator: J. Lukeš, **2017-2021**.
* Russian Science Foundation grant 14-14-00988 “The function and evolution of CRISPR/Cas system”. Principal investigator: K. Severinov, **2014-2017**.

Conferences
------

* **Nenarokova A.**, Záhonová K., Nenarokov S., Paris Z., Yurchenko V., Lukeš J. *Blastocrithidia*, the trypanosomatid with all three stop codons reassigned. // Joint meeting of the Phycological Society of America and the International Society of Protistologists, Vancouver, Canada, 2018. Oral presentation
* **Nenarokova A.**, Záhonová K., Nenarokov S., Yurchenko V., Lukeš J. Genomics of *Blastocrithidia*, the trypanosomatid with all three stop codons reassigned. // XXII Meeting of the International Society for Evolutionary Protistology, Droushia, Cyprus, 2018. Oral presentation
* **Nenarokova A.**, Záhonová K., Nenarokov S., Paris Z., Boudová M.,  Skalický T., Yurchenko V., Lukeš J. *Blastocrithidia*, a trypanosomatid with all three stop codons reassigned. // EMBO meeting, Sant Feliu de Guixols, Spain, 2017. Poster. **Best Poster Award**
* **Nenarokova A.**, Záhonová K., Nenarokov S., Paris Z., Boudová M.,  Skalický T., Yurchenko V., Lukeš J. *Blastocrithidia*, a trypanosomatid with all three stop codons reassigned. // International Congress of Protistologists, Prague, Czech Republic, 2017. Oral presentation
* **Nenarokova A.**, Dobáková E., Flegontov P., Ebenezer T., Zolner M., Field M., Lukeš J. Functional annotation of *Euglena gracilis* mitochondrial proteome. // British Society for Parasitology Trypanosomiasis & Leishmaniasis Seminar, České Budějovice, Czech Republic, 2016. Poster
* **Nenarokova A.**, Flegontov P., Field  M.C., Lukeš J. Functional annotation of *Euglena gracilis* mitochondrial proteome. // TryTax2 Meeting, Ostrava, Czech Republic, 2016. Poster
* **Nenarokova A.**, Flegontov P., Field  M.C., Lukeš J. Functional annotation of *Euglena gracilis* mitochondrial proteome. //  Protodays, Bítov, Czech Republic, 2016. Poster. **Best Poster Award**
* **Nenarokova A.A.** Polymorphism analysis of NBS LRR class resistance genes in wheat varieties using high-throughput sequencing. // Bioinformatics Summer School, St. Petersburg, Russia, 2015. Oral presentation
* **Nenarokova A.A.**, Strotskaya A.V. Adaptation process in the CRISPR immunity system. // The 18th Pushchino School-Conference of Young Scientists “Biology – The Science of The XXI Century”, Pushchino, Russia, 2014. Poster. **Best Poster Award**
* **Nenarokova A.A.** Experimental system for studying the Effects of CRISPR/Cas genetic immunity system of *Escherichia coli* against lytic bacteriophages. //XXI Undergraduate and Postgraduate Student International Conference "Lomonosov",  Moscow, Russia, 2014. Poster

Attended Workshops
-----

* Workshop on phylogenomics [evomics.org](http://www.evomics.org), Český Krumlov, Czech Republic, 2017
* Elixir Excelerate workshop, Prague, 2016
* III Bioinformatics Summer School, St. Petersburg, Russia, 2015. Project: “Analysis of *Homo sapiens* Reference Genome”
* Future Biotech Winter School, Zvenigorod, Russia, 2015
* Future Biotech Autumn Intensity, Moscow, Russia, 2014
* II Bioinformatics Summer School, St. Petersburg, Russia, 2014. Project: “Genomic Analysis of Pathogenic Regions in *E. coli* O104:H4”

Teaching
-----

* Organizer and lecturer of the “Introduction to programming for biologists” seminar at the Institute of Parasitology, Biology Centre CAS; ongoing.
* Teaching assistant to the “Introduction to Bioinformatics” course at the University of South Bohemia; 2018.
* Bioinformatics mentoring for biologists; 2017-present

Skills
-----

* Administering Linux workstations and clusters
* Programming: scripting (Bash, Python, R); reading and modifying code (C)
* Preparing of raw NGS reads: debarcoding, trimming, merging
* Genome and transcriptome assembly
* Read mapping and polymorphism analysis
* Structural and functional annotation of prokaryotic and eukaryotic genomes
* Genome alignment and comparative genomics methods
* Phylogenetics and phylogenomics analysis

Languages
-----

* **Russian** native
* **English** full proficiency (C1)
* **Czech** conversational, reading professional literature (A2)
* **French** reading scientific literature

Research Interests
------

* Origin and evolution of eukaryotes
* Protist genomics
* Evolution of the genetic code
